extensions/amp-intaker-chat-widget/0.1/widget.js

```diff
@@ -0,0 +1,570 @@
+          a.substr(0, 4)
+        )
+      ) {
+        check = true;
+      }
+    })(navigator.userAgent || navigator.vendor || window.opera);
+    return check;
+  }
+
+  /**
+   *
+   * @param {function(): undefined} callback
+   */
+  function authenticate(callback) {
+    // if (useQA)
+    //   return callback();
+
+    postAjax(
+      api + '/api/Chat/CheckCustomerSubscription',
+      /** @type {!JsonObject} */ ({
+        'directLink': directLink,
+        'externalLink': externalUrl,
+      }),
+      function(result) {
+        result = Intaker.parseJson(result);
+        if (result === 'Paid' || result === 'Trial') {
+          callback();
+        } else {
+          if (console.error) {
+            if (result === 'CustomerNotFound') {
+              console /*OK*/
+                .error('This is not the chat you are looking for.');
+            } else {
+              console /*OK*/
+                .error(
+                  'Intaker Chatter Bot client is expired. Login to your dashboard and upgrade.'
+                );
+              console /*OK*/
+                .error('https://intaker.co/dashboard');
+            }
+          }
+        }
+      }
+    );
+  }
+
+  /**
+   *
+   * @param {Object} setting
+   */
+  function injectThemeCss(setting) {
+    const avatar = setting.avatarUrl || DEFAULT_AVATAR;
+    const color = setting.colorPick || '2DC464';
+    const css =
+      '#chatter-bot-launcher-container .chatter-bot-launcher-button {background-image : url(' +
+      avatar +
+      ');}' +
+      '#chatter-bot-launcher {background : #' +
+      color +
+      ';}';
+    const linkElement = document.createElement('link');
+    linkElement.setAttribute('rel', 'stylesheet');
+    linkElement.setAttribute('type', 'text/css');
+    linkElement.setAttribute(
+      'href',
+      'data:text/css;charset=UTF-8,' + encodeURIComponent(css)
+    );
+    document.body.appendChild(linkElement);
+  }
+
+  /**
+   *
+   * @param {function(?): undefined} callback
+   */
+  function getChatSetting(callback) {
+    postAjax(
+      api + '/api/Chat/GetChatSetting',
+      /** @type {!JsonObject} */ ({
+        'directLink': directLink,
+        'externalLink': externalUrl,
+      }),
+      function(result) {
+        result = Intaker.parseJson(result);
+        injectThemeCss(result);
+        callback(result);
+      }
+    );
+  }
+
+  /**
+   *
+   */
+  function playDingSound() {
+    const promise = dingSound.play();
+    if (promise !== undefined) {
+      promise.then(function() {}).catch(function() {});
+    }
+  }
+
+  /**
+   *
+   */
+  function loadDingSound() {
+    if (window.Audio) {
+      dingSound = new Audio();
+      dingSound.src = 'https://intaker.co/dist/light.mp3';
+      dingSound.load();
+    }
+  }
+
+  /**
+   *
+   * @param {Object} setting
+   * @return {undefined}
+   */
+  function autoLunch(setting) {
+    const cookie = CookiesAPI.getJSON(cookieName);
```

Comment:
Why make these calls if the "API" is implemented as empty?

Reply:
As far as I know there is no cookie support in AMP! right? 
This is a fully working widget in native js for normal sites and I did not wanted to maintain 2 different versions for normal sites and AMP sites, so I made some changes to make it work with AMP, in normal sites CookiesAPI is loaded from customized [https://github.com/js-cookie/js-cookie](url). 

I added the empty CookiesAPI code to make the widget work as is.

Reply:
updated, now using the cookie module, I am gana need a Privacy Review